pkg/reconciler/ingress/status.go

```diff
@@ -323,15 +325,27 @@ func (m *StatusProber) listVirtualServicePodIPs(vs *v1alpha3.VirtualService) ([]
 			}
 			selector = selector.Add(*requirement)
 		}
-		pods, err := m.podLister.List(selector)
+
+		endpoints, err := m.endpointsLister.List(selector)
 		if err != nil {
-			return nil, fmt.Errorf("failed to list Pods: %v", err)
+			return nil, fmt.Errorf("failed to list Endpoints: %v", err)
 		}
 
-		// Filter out the Pods without an assigned IP address
-		for _, pod := range pods {
-			if pod.Status.PodIP != "" {
-				podIPs = append(podIPs, pod.Status.PodIP)
+		for _, eps := range endpoints {
+			for _, sub := range eps.Subsets {
+				http2Port := ""
+				for _, port := range sub.Ports {
+					if port.Name == "http2" {
```

Comment:
I think a comment about where this constant comes from would be useful.

Reply:
Done.